test/integration/variable_test.rb

```diff
@@ -130,4 +130,8 @@ def test_dynamic_find_var
   def test_raw_value_variable
     assert_template_result('bar', '{{ [key] }}', { 'key' => 'foo', 'foo' => 'bar' })
   end
+
+  def test_hash_last
+    assert_template_result('last', '{{ hash.last }}', { 'hash' => { 'first' => 'first', 'last' => 'last' } })
```

Comment:
Whoops, this test actually doesn't exercise the new path because the key is also `last`. Fixing now.